src/Main.elm

```diff
@@ -0,0 +1,159 @@
+module Main exposing (Model, Msg, init, update, view)
+
+import Browser
+import Colors exposing (black, blue, blueGreen, lightBlue, orange, purple, sky, white)
+import Element exposing (centerX, centerY, column, el, fill, height, html, layout, maximum, padding, rgb255, shrink, spacing, text, width)
+import Element.Background as Background
+import Element.Border as Border
+import Element.Font as Font exposing (center)
+import Element.Input as Input exposing (button)
+import Http
+import Json.Decode as JD
+import List.Zipper exposing (Zipper, fromCons)
+import Maybe
+import Ranking
+import RemoteData
+import SetZipper exposing (RankedTeam, TeamRankings)
+import Setup exposing (Model)
+import String
+
+
+main : Program () Model Msg
+main =
+    Browser.element
+        { init = always init
+        , view = view >> layout []
+        , update = update
+        , subscriptions = always Sub.none
+        }
+
+
+type Pages
+    = Setup
+    | Ranking
+
+
+type Msg
+    = RankingM Ranking.Msg
+    | SetupM Setup.Msg
+    | Switch
+    | GotRankings (RemoteData.RemoteData Http.Error TeamRankings)
+
+
+type alias Model =
+    { setup : Setup.Model
+    , ranking : Ranking.Model
+    , pages : Pages
+    , rankings : RemoteData.RemoteData Http.Error TeamRankings
+    }
+
+
+type alias TeamRankings =
+    List RankedTeam
+
+
+type alias RankedTeam =
+    { name : String
+    , position : Int
+    }
+
+
+switchButton : String -> Element.Element Msg
+switchButton string =
+    el
+        [ Element.centerX
+        , Element.centerY
+        , Element.moveDown 40
+        , width fill
+        , height fill
+        , Font.size 40
+        ]
+    <|
+        Input.button
+            [ Border.rounded 8
+            , Background.gradient
+                { angle = 2
+                , steps = [ purple, orange, blueGreen ]
+                }
+            , center
+            , Element.centerX
+            , Element.centerY
+            ]
+            { onPress = Just Switch
+            , label = text string
+            }
+
+
+rankingParser : JD.Decoder TeamRankings
+rankingParser =
+    JD.map2 RankedTeam (JD.field "team_key" JD.string) (JD.field "rank" JD.int)
+        |> JD.list
+        |> JD.field "rankings"
+
+
+init : ( Model, Cmd Msg )
+init =
+    ( { setup = Setup.init 1 1 1 1 2017 2020
+      , pages = Setup
+      , rankings = RemoteData.Loading
+      , ranking = Ranking.init RemoteData.Loading
```

Comment:
> In Main.elm in function init i have this line
> ranking = Ranking.init RemoteData.Loading
> I know its wrong, but I need somehow send the info from Tba in the init.
> That I can not achieve because I need the model.rankings for this

What's wrong about this line?
About TBA, you send a GET request to TBA right here so I don't see a problem.